### ISSUE TITLE
[Backport ] Deletion of downstream RKE2 cluster with CIS1.6 profile - cattle-cleanup-c5s6d-g44cr stuck in CreateContainerConfigError

### ISSUE THREAD

**Issue by rancherbot** [rancher/rancher#39785]
This is a backport issue for https://github.com/rancher/rancher/issues/33690, automatically created via rancherbot by @sbstp

Original issue description:

**Rancher Server Setup**
- Rancher version: 2.5.8
- Installation option (Docker install/Helm Chart): helm chart on rke2
- Proxy/Cert Details: cert-manager self signed CA

**Information about the Cluster**
- Kubernetes version: 1.20/1.21
- Cluster Type (Local/Downstream): imported RKE2

**Describe the bug**
Deletion of the downstream cluster in Rancher UI creates a pod in the downstream cluster that is not valid for the given pod security policies:

```
d050vlcsm801:~ # kubectl describe pods cattle-cleanup-c5s6d-g44cr
Name:         cattle-cleanup-c5s6d-g44cr
Namespace:    default
Priority:     0
Node:         d050vlcsw804/53.190.243.198
Start Time:   Tue, 20 Jul 2021 13:47:52 +0000
Labels:       controller-uid=4798ff27-bc5f-487b-81b1-9a66b33cc9c2
              job-name=cattle-cleanup-c5s6d
Annotations:  cni.projectcalico.org/podIP: 192.168.3.7/32
              cni.projectcalico.org/podIPs: 192.168.3.7/32
              kubernetes.io/psp: global-restricted-psp
```

And due to this it fails in CreateContainerConfigError with

` Error: container has runAsNonRoot and image will run as root (pod: "cattle-cleanup-c5s6d-g44cr_default(a8fd73c2-fd53-4520-aa56-061f38efa267)", container: cleanup-agent)
`

**To Reproduce**
Import RKE2 downstream cluster that was configured with CIS profile 1.5 or 1.6
Then delete the cluster from the UI and find the stuck pod in the default namespace of the downstream cluster.

**Result**
Stuck pod

**Expected Result**
No stuck pod

SURE-3194

**Comment by markusewalker** [comment]
**QA TEST PLAN**

**Scenarios**
|Scenario|Test Case|
|---|---|
|1|As a standard user, import an RKE2 hardened cluster into a fresh install of Rancher v2.6-head, delete the cluster in Rancher, validate that there are no failing pods|
|2|Repeat scenario 1, but initially setup Rancher v2.6.9 and do not delete the cluster. Upgrade Rancher to v2.6-head, validate the cluster and nodes are still active and then delete the cluster in Rancher. Validate that there are no failing pods|

**Comment by markusewalker** [comment]
Reproduced this on `v2.6.9`. Will attempt to validate with the original [test plan](https://github.com/rancher/rancher/issues/39785#issuecomment-1343195289), but find repro details below:

**ENVIRONMENT DETAILS**
- Rancher Install: Helm
- Rancher version: `v2.6.9`
- Browser: Chrome

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|As a standard user, import an RKE2 hardened cluster into a fresh install of Rancher v2.6.9, delete the cluster in Rancher, validate that there are no failing pods|:x:|

**REPRO STEPS**

**Scenario 1**
1. On a separate terminal, created an RKE2 hardened cluster with CIS1.6 profile.
2. Setup Rancher, created a standard user and logged in as that user.
3. Imported the previously created RKE2 hardened cluster.
4. Validated that the cluster and nodes came up as `Active` in Cluster Management and Cluster Explorer.
5. Deleted the cluster from Rancher.
6. Navigated back to a terminal where the cluster's kubeconfig lives and ran command `kubectl get pods -A`.
7. Validated that there IS an error with the cattle-cleanup pod:
```
# kubectl get pods -A
NAMESPACE             NAME                                                    READY   STATUS                       RESTARTS   AGE
cattle-fleet-system   fleet-agent-bfc5655cc-jrwxk                             1/1     Running                      0          57s
cattle-system         cattle-cluster-agent-dcd6bfb4b-r4cvp                    1/1     Running                      0          88s
cattle-system         cattle-cluster-agent-dcd6bfb4b-wqwks                    1/1     Running                      0          117s
default               cattle-cleanup-f9cbd-d26t2                              0/1     CreateContainerConfigError   0          36s
```

**Comment by markusewalker** [comment]
Validated that this is addressed in `v2.6-head`  https://github.com/rancher/rancher/commit/8048eee359be56f3419f37e1956a2e2ecf57aa53. Details below:

- Rancher Install: Helm
- Rancher version: `v2.6-head` https://github.com/rancher/rancher/commit/8048eee359be56f3419f37e1956a2e2ecf57aa53
- Browser: Chrome

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|As a standard user, import an RKE2 hardened cluster into a fresh install of Rancher v2.6-head, delete the cluster in Rancher, validate that there are no failing pods|:white_check_mark:|
|2|Repeat scenario 1, but initially setup Rancher v2.6.9 and do not delete the cluster. Upgrade Rancher to v2.6-head, validate the cluster and nodes are still active and then delete the cluster in Rancher. Validate that there are no failing pods|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. On a separate terminal, created an RKE2 hardened cluster with CIS1.6 profile.
2. Setup Rancher, created a standard user and logged in as that user.
3. Imported the previously created RKE2 hardened cluster.
4. Validated that the cluster and nodes came up as `Active` in Cluster Management and Cluster Explorer.
5. Deleted the cluster from Rancher.
6. Navigated back to a terminal where the cluster's kubeconfig lives and ran command `kubectl get pods -A`.
7. Validated that there are no pods with an error and cleanup was successful:
```
# kubectl get pods -A
NAMESPACE     NAME                                                    READY   STATUS      RESTARTS   AGE
kube-system   cloud-controller-manager-ip-172-31-19-223               1/1     Running     8          6m21s
kube-system   etcd-ip-172-31-19-223                                   1/1     Running     5          5m49s
kube-system   helm-install-rke2-canal-j4xk6                           0/1     Completed   0          6m2s
kube-system   helm-install-rke2-coredns-qtsxj                         0/1     Completed   0          6m2s
kube-system   helm-install-rke2-ingress-nginx-85gkw                   0/1     Completed   0          6m1s
kube-system   helm-install-rke2-metrics-server-knhkz                  0/1     Completed   0          6m1s
kube-system   kube-apiserver-ip-172-31-19-223                         1/1     Running     6          6m23s
kube-system   kube-controller-manager-ip-172-31-19-223                1/1     Running     8          6m23s
kube-system   kube-proxy-ip-172-31-19-223                             1/1     Running     5          6m1s
kube-system   kube-proxy-ip-172-31-22-195                             1/1     Running     3          4m49s
kube-system   kube-proxy-ip-172-31-28-67                              1/1     Running     3          5m3s
kube-system   kube-scheduler-ip-172-31-19-223                         1/1     Running     6          6m22s
kube-system   rke2-canal-46c68                                        2/2     Running     0          5m55s
kube-system   rke2-canal-j95dd                                        2/2     Running     0          5m4s
kube-system   rke2-canal-vmhzn                                        2/2     Running     0          5m2s
kube-system   rke2-coredns-rke2-coredns-58fd75f64b-kzb5p              1/1     Running     0          5m55s
kube-system   rke2-coredns-rke2-coredns-58fd75f64b-zltsq              1/1     Running     0          5m
kube-system   rke2-coredns-rke2-coredns-autoscaler-768bfc5985-x9lhj   1/1     Running     0          5m55s
kube-system   rke2-ingress-nginx-controller-c7g85                     1/1     Running     0          5m24s
kube-system   rke2-ingress-nginx-controller-dqxv8                     1/1     Running     0          4m44s
kube-system   rke2-ingress-nginx-controller-h5zv6                     1/1     Running     0          4m41s
kube-system   rke2-metrics-server-67697454f8-67pqs                    1/1     Running     0          5m31s
```

___________
**Scenario 2**
1. Repeated scenario 1, but with Rancher 2.6.9 and ensured that I did not delete the cluster.
2. Upgraded Rancher to `v2.6-head` https://github.com/rancher/rancher/commit/8048eee359be56f3419f37e1956a2e2ecf57aa53; validated that the cluster did not auto-upgrade K8s version and that it remained `Active`.
3. Deleted the cluster from Rancher.
4. Navigated back to a terminal where the cluster's kubeconfig lives and ran command `kubectl get pods -A`.
5. Validated that there are no pods with an error and cleanup was successful.
